imagebuildah/stage_executor.go

```diff
@@ -1019,22 +1042,16 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			if imgID, ref, err = s.tagExistingImage(ctx, s.builder.FromImageID, s.output); err != nil {
 				return "", nil, err
 			}
-			if s.executor.buildOutput != "" && lastStage {
-				// If we have reached this point then our build is just performing a tag
-				// and it contains no steps or instructions (i.e Containerfile only contains
-				// `FROM <imagename> and nothing else so we will never end up committing this
-				// but instead just re-tag image. For such use-cases if `-o` or `--output` was
-				// specified honor that and export the contents of the current build anyways.
-				logrus.Debugf("Generating custom build output with options %q", s.executor.buildOutput)
-				buildOutputOption, err := parse.GetBuildOutput(s.executor.buildOutput)
-				if err != nil {
-					return "", nil, fmt.Errorf("failed to parse build output: %w", err)
-				}
-				if err := s.generateBuildOutput(buildah.CommitOptions{}, buildOutputOption); err != nil {
+			// If we have reached this point then our build is just performing a tag
+			// and it contains no steps or instructions (i.e Containerfile only contains
+			// `FROM <imagename> and nothing else so we will never end up committing this
+			// but instead just re-tag image. For such use-cases if `-o` or `--output` was
+			// specified honor that and export the contents of the current build anyways.
```

Comment:
I'm a little confused by this comment and could just be missing it.  You say in the comment that if the -o or --output option was specified, then we should generate output.  I don't see a check for either option below?  I don't think checking either of those options is part of the canGenerateBuildOutput boolean either.

Reply:
Oh i see but the comment was already there I just moved it one line above and comment looks still true to me let me know if you think we need to repharse, the branch where we have this check ensures that we are just retagging and will not commit so we must consider generating the output if user specified `--output`, `canGenerateBuildOutput ` ensures if `--output` was specified or not.

Reply:
Actual change was added here: https://github.com/containers/buildah/pull/4097